greenery/fsm_test.py

```diff
@@ -421,14 +423,14 @@ def test_binary_3():
     assert div3.accepts("1001")
 
 
-def test_invalid_fsms():
+def test_invalid_fsms() -> None:
     # initial state 1 is not a state
     with pytest.raises(Exception, match="Initial state"):
         Fsm(
             alphabet={},
             states={},
             initial=1,
-            finals=set(),
+            finals=(),
```

Comment:
This must be new syntax since last time I checked how to create an empty `set`, nice.

Reply:
It's actually just an empty tuple, accepted since `Fsm` now accepts any `Iterable`. The minor general-practice benefit vs. `set()` is that it can sometimes avoid allocations (`() is ()` but `set() is not set()`) and it signals that refactoring/sharing/caching is ok for that value, since tuples are immutable. Neither is important here, it’s just nicer and very slightly briefer.